habitat_baselines/rl/ppo/ppo_trainer.py

```diff
@@ -350,69 +338,22 @@ def train(self) -> None:
                         logger.info("No episodes finish in current window")
 
                 # checkpoint model
-                if update % ppo_cfg.checkpoint_interval == 0:
+                if update % self.config.CHECKPOINT_INTERVAL == 0:
                     self.save_checkpoint(f"ckpt.{count_checkpoints}.pth")
                     count_checkpoints += 1
 
-    def eval(self) -> None:
-        r"""Main method of evaluating PPO.
-
-        Returns:
-            None
-        """
-        ppo_cfg = self.config.TRAINER.RL.PPO
-        self.device = torch.device("cuda", ppo_cfg.pth_gpu_id)
-        self.video_option = ppo_cfg.video_option.strip().split(",")
-
-        if "tensorboard" in self.video_option:
-            assert (
-                ppo_cfg.tensorboard_dir is not None
-            ), "Must specify a tensorboard directory for video display"
-        if "disk" in self.video_option:
-            assert (
-                ppo_cfg.video_dir is not None
-            ), "Must specify a directory for storing videos on disk"
-
-        with get_tensorboard_writer(
-            ppo_cfg.tensorboard_dir, purge_step=0, flush_secs=30
-        ) as writer:
-            if os.path.isfile(ppo_cfg.eval_ckpt_path_or_dir):
-                # evaluate singe checkpoint
-                self._eval_checkpoint(ppo_cfg.eval_ckpt_path_or_dir, writer)
-            else:
-                # evaluate multiple checkpoints in order
-                prev_ckpt_ind = -1
-                while True:
-                    current_ckpt = None
-                    while current_ckpt is None:
-                        current_ckpt = poll_checkpoint_folder(
-                            ppo_cfg.eval_ckpt_path_or_dir, prev_ckpt_ind
-                        )
-                        time.sleep(2)  # sleep for 2 secs before polling again
-                    logger.warning(
-                        "=============current_ckpt: {}=============".format(
-                            current_ckpt
-                        )
-                    )
-                    prev_ckpt_ind += 1
-                    self._eval_checkpoint(
-                        checkpoint_path=current_ckpt,
-                        writer=writer,
-                        cur_ckpt_idx=prev_ckpt_ind,
-                    )
-
     def _eval_checkpoint(
```

Comment:
The modularity for eval is also looking better, one suggestion can we split the `_eval_checkpoint` into further modular steps, like in 1) construction of env, 2) evaluation inside envs.